test/tests/adaptivity/dont-p-refine/tests

```diff
@@ -22,71 +22,71 @@
       input = test.i
       csvdiff = test_out.csv
       prereq = 'disable_p_refinement/lagrange_second_order_mesh'
-      cli_args = 'Adaptivity/disable_p_refinement_for_families="hermite" AuxVariables/test/family=HERMITE AuxVariables/test/order=THIRD'
+      cli_args = 'Mesh/disable_p_refinement_for_families="hermite" AuxVariables/test/family=HERMITE AuxVariables/test/order=THIRD'
```

Comment:
yea, I'm not a fan of this at all. What's the point of having an `Adaptivity` block at this point

Reply:
The mesh could be p-refined by loading a checkpoint file. In such a case, we do not have `[Adpativity]` but we still need to disable certain shape function families. I think this is the reason that we need to move this into `[Mesh]`. We could move this into `[Problem]`, but that is a syntax used for creating custom problems and may create implications. Having this parameter in `[Mesh]` just mean that the created mesh will not support these shape functions. Most of times, users can just use the default I guess.

Reply:
A mesh has absolutely no data about finite element families

Reply:
It might be possible to move the code directly in `FEProblemBase::init` at its beginning. If so, `disable_p_refinement_for_families` can be a parameter of `[Problem]`.

Reply:
If it is restartable data, it won't be available before calling `FEProblemBase::init`. Will that be too late?

Reply:
I don't think it would be too late, but I am not 100% sure. It just needs to happen before a `FE::reinit` occurs. I would like to at least try and see becuase I still think this parameter is a better fit in the `Adaptivity` block as it is about adaptivity